exporter/elasticsearchexporter/factory.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package elasticsearchexporter
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "elasticsearch"
+
+	mappingFieldName = "mapping"
+)
+
+// NewFactory creates a factory for Elastic exporter.
+func NewFactory() component.ExporterFactory {
+	return exporterhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		exporterhelper.WithLogs(createLogsExporter),
+	)
+}
+
+func createDefaultConfig() configmodels.Exporter {
+	return &Config{
+		ExporterSettings: configmodels.ExporterSettings{
+			TypeVal: configmodels.Type(typeStr),
+			NameVal: typeStr,
+		},
+		HTTPClientSettings: HTTPClientSettings{
+			Timeout: 90 * time.Second,
```

Comment:
Is this a reasonable default?

Reply:
It depends. I was considering to make timeouts more granular, e.g. a connection timeout 
(like 5s) and introduce a ResponseHeaderTimeout of 10s.

As part of congestion control Elasticsearch does report 429. Actually Elasticsearch uses selective ACK, telling us exactly which events could not have been indexed and need to be retried. That is there is some timeout server-side. In order to reduce bandwidth usage and reduce the chance of duplicates we want these per document status codes.

Users that have configured a pipeline or having a template with a default pipeline might also want to increase the timeout, due to the additional document processing times. All in all I opted to be a little more generous here, in order to reduce the need to tune the setting due to reoccurring timeouts.

Reply:
Alright. In general cases, I would say that a couple of seconds is already high enough, but if 90s is within the reasonable for this use case, it's fine by me.